src/Control/Monad/HashCons.hs

```diff
@@ -101,6 +112,7 @@ hashCons x = HashConsT $ do
 -- 2. @'liftA2' (/=) ('refTo' x) ('refTo' y)@ @=@
 --    @'liftA2' \\(idX idX -> (x '/=' y) '==' (idX '/=' idY)) ('refTo' x)
 --    ('refTo' y)@
+-- 3. @'refTo' x '>>=' 'lookupRef'@ @=@ @'pure' (Just x)@
```

Comment:
This is technically not possible. `=` is a rewrite, and rewriting any use of `refTo` into `pure` cannot be correct, as `refTo` has effects. At the very least, `HashConsT` does not satisfy this law as it is currently phrased.